### ISSUE TITLE
diagnostic(p3): re-derive the 308 random baseline under matched conditions

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/bucket-brigade#192]
## Hypothesis (H3)

**The \"308 random baseline\" used in #174 / #183 acceptance criteria was not computed under conditions matching the training cells.** If the baseline is wrong, the framing "PPO is below random" (which currently sits at iter-0 reward 290.5 from the #183 phase 3 L1_norm cell) may be misleading.

## Why this matters

A core finding driving the H1–H4 discussion is: **iter-0 PPO reward (290) < random baseline (308)**, suggesting random-init PPO is worse than uniform-random actions. This is a striking claim. Before accepting it, verify the baseline was computed correctly.

Two readings of the 290 vs 308 gap:

- (a) Baseline was apples-to-apples → random-init MLP IS worse than uniform-random, very suspicious finding
- (b) Baseline was different scenario / different episode count / uniform-actions vs random-MLP-init → 290 IS effectively random and PPO simply isn't learning (consistent with H1/H2 without requiring "actively worse than random")

## What to measure

Run a uniform-random-action policy through the **same** env/scenario/episode-count used in #183 phase 3:

- Scenario: \`default\`
- 4 agents
- Episode count: enough rollouts to match the per-iteration \`rollout_steps=2048\` × \`num_iterations=500\` = ~1M env steps in #183, but for the baseline ~50 episodes × episode_length is plenty
- Action policy: sample uniformly from the action space (\`a[:, 0]\` ∈ [0..9], \`a[:, 1]\` ∈ [0..1]) at every step

Compare:

- Uniform-random mean per-step team reward
- Random-init PPO MLP mean per-step team reward (iter 0 of any #183 cell)
- The "308" cited as the baseline in #174 / #183 — where did this number come from?

## Acceptance criteria

- [ ] Uniform-random mean per-step team reward reported (mean ± 95% CI over 50+ episodes)
- [ ] Random-init MLP mean per-step team reward at iter 0 reported (already have this from #183 cells: 290.52 for L1_norm, similarly for L1L2/L1L3)
- [ ] Provenance of "308" identified — locate the script/notebook that produced it and confirm the conditions
- [ ] Verdict: do all three numbers agree (random is ~290 or ~308 across conditions), or is there a gap requiring explanation?

## Implementation notes

- A standalone 30-line script in \`experiments/p3_specialization/diagnostics/\` is the right deliverable
- Use \`BucketBrigadeEnv\` with \`default_scenario(num_agents=4)\` (or whatever \`default\` resolves to via \`get_scenario_by_name\`)
- Reset with a few different seeds (e.g., 42, 43, 44) to bound variance
- The "308 random baseline" likely came from a #145 or earlier diagnostic — \`grep -rn "308\\|random.*baseline" research_notebook/ experiments/\` to find provenance

## Relationship to H1 / H2 / H4

- If 308 is right and 290 is real: PPO is **actively** harming itself relative to random. Strong evidence for either H1 (no signal) or H2 (team-dominated, anti-collaborative gradient).
- If the true random baseline is ~290: PPO is at random and stays there. Consistent with H1/H2 but less dramatic.
- Either way: H3 confirms or weakens the framing for whatever intervention H4 lands on.

## Out of scope

- Re-running PPO training
- Computing a "fixed-good-strategy" baseline (e.g., a hand-coded policy that just works hard every night)
- Other scenarios — start with default, expand if useful

## Estimated effort

~20 min. Cheapest of the diagnostic issues but contingent on H1/H2 to interpret.

**Comment by rjwalters** [comment]
## Curator Enhancement

### Provenance of "308"

**Origin: issue #145** (closed). The number was measured and reported in the issue body:

> Random actions across 50 episodes on `default`: 4012.34 per episode (~+308/step).

That `308 = 4012.34 / ~13` implies an episode length of ~13 nights — consistent with `default_scenario`'s `min_nights=12` plus the termination check that allows the game to continue past `min_nights` if fires are still active (`bucket_brigade/envs/bucket_brigade_env.py:303–314`). The script/notebook that produced 4012.34 is **not in the repo** — the issue says "random actions across 50 episodes" but no committed script reproduces it. This is itself a reason to re-derive.

The 308 is then hard-coded in two places:
- `experiments/p3_specialization/analyze_plateau.py:48–52` (`BASELINES` dict, with `"default": {"random": 308.0, "heuristic": 307.0}`)
- `experiments/p3_specialization/analyze_174.py:47` (`RAND_BASELINE = 308.0`)

It is then transitively cited in:
- `experiments/p3_specialization/diagnostics/summary.md:16` (293.46 → 294.58 vs baseline 308)
- `experiments/p3_specialization/results_174_ablation/summary.md:3` (acceptance bar > 320)
- `research_notebook/2026-05-14_p3_specialization_results.md:92`
- `research_notebook/2026-05-15_p3_plateau_ablation.md:14,32,50,63`

**Conditions per #145 body**: `default` scenario, 4 agents implied (P3 default), 50 episodes, uniform-random actions. Episode length is whatever `min_nights=12` + early-termination chance produces (~13 nights). These conditions **match** the #183 training cells (same scenario, 4 agents, `default_scenario`). So if there's a discrepancy between 290 and 308, it's not from scenario/agent-count mismatch — most likely:
- (a) sample noise (only 50 episodes on a high-variance reward),
- (b) per-step vs per-episode normalization (4012.34 / 13 ≈ 308.6 vs 4012.34 / 12 ≈ 334.4 — sensitive to which divisor was used),
- (c) random-init MLP is genuinely slightly below uniform-random (the policy starts biased due to Xavier init + softmax on logits that won't be perfectly uniform across the multi-discrete dims).

### Env API verification (matches issue body, with one correction)

`bucket_brigade/envs/bucket_brigade_env.py:117` documents `actions` as **shape `(N, 2)` with `[house_index, mode_flag]`**:
- `actions[:, 0]` ∈ `[0..9]` (house index — 10 houses)
- `actions[:, 1]` ∈ `[0..1]` (mode flag: 0=rest, 1=work)

Confirmed in `bucket_brigade/envs/puffer_env.py:77`: `single_action_space = gym_spaces.MultiDiscrete([10, 2])`.

**Correction to issue body**: episode length is **not "13 nights" by spec** — it is `min_nights=12` followed by termination when no fires are burning / all houses safe or ruined. Empirical episode length will be 12 or slightly more depending on fire dynamics. Use the actual observed length (record `info["night"]` at done) for per-step normalization rather than assuming a fixed number.

### Matched-conditions baseline spec

| Parameter | Value | Source |
|---|---|---|
| Scenario | `default` via `get_scenario_by_name("default", num_agents=4)` | `bucket_brigade/envs/scenarios_generated.py:70,305` |
| `num_agents` | 4 | matches P3 sweep `train.py:63`, #183 cells |
| Action sampling | `np.random.randint(0, 10)` for dim 0, `np.random.randint(0, 2)` for dim 1, per agent per step | `MultiDiscrete([10, 2])` |
| Episodes per seed | ≥ 200 (push 50 → 200+ to tighten CI) | original was 50 |
| Seeds | 5+ (e.g., 42–46) for between-seed variance bound | original was unstated |
| Metric | per-step team reward = `rewards.sum() / episode_length`, where `episode_length` = actual nights played | matches #145 normalization |
| Reporting | mean ± 95% bootstrap CI over `n_episodes × n_seeds` samples | matches paper's statistical protocol |

### Diagnostic script outline (~30 LOC)

```python
# experiments/p3_specialization/diagnostics/random_baseline.py
import numpy as np
from bucket_brigade.envs import BucketBrigadeEnv
from bucket_brigade.envs.scenarios_generated import get_scenario_by_name

def run_random_episode(scenario, rng):
    env = BucketBrigadeEnv(scenario=scenario, seed=int(rng.integers(0, 2**31)))
    env.reset()
    total_reward, steps = 0.0, 0
    while not env.done:
        actions = np.stack([
            rng.integers(0, 10, size=scenario.num_agents),
            rng.integers(0, 2, size=scenario.num_agents),
        ], axis=-1)  # shape (N, 2)
        _, rewards, _, _ = env.step(actions)
        total_reward += rewards.sum()  # team reward = sum over agents
        steps += 1
    return total_reward, steps

def main():
    scenario = get_scenario_by_name("default", num_agents=4)
    per_step_rewards = []
    for seed in (42, 43, 44, 45, 46):
        rng = np.random.default_rng(seed)
        for _ in range(200):
            ep_reward, ep_steps = run_random_episode(scenario, rng)
            per_step_rewards.append(ep_reward / ep_steps)
    arr = np.array(per_step_rewards)
    # 95% bootstrap CI
    boots = [np.random.choice(arr, len(arr), replace=True).mean() for _ in range(10000)]
    lo, hi = np.percentile(boots, [2.5, 97.5])
    print(f"Uniform-random per-step team reward: mean={arr.mean():.2f}, 95% CI=[{lo:.2f}, {hi:.2f}], n={len(arr)} episodes")
    print(f"Median episode length: {np.median([...])}")  # add length tracking

if __name__ == "__main__":
    main()
```

### Acceptance Criteria

- [ ] Report uniform-random per-step team reward on `default` w/ `num_agents=4`: **mean ± 95% bootstrap CI over ≥ 1000 episodes (≥ 5 seeds × ≥ 200 episodes)**
- [ ] Report random-init MLP iter-0 per-step team reward from any #183 phase-3 cell (already available: 290.52 for L1_norm; collect others)
- [ ] Compare both to the cited **308** from #145
- [ ] Verdict: do all three agree within CI? If not, identify which is the outlier and why (sample size, normalization, random-init MLP bias)
- [ ] If a discrepancy is found, update the H1/H2/H4 framing accordingly (i.e., if the true random baseline is ~290, then "PPO at random" is the correct framing, not "PPO worse than random")

### Affected Files

- `experiments/p3_specialization/diagnostics/random_baseline.py` (NEW — the diagnostic script)
- `experiments/p3_specialization/diagnostics/summary.md` (UPDATE — add re-derived baseline)
- `research_notebook/2026-05-15_p3_plateau_ablation.md` (UPDATE — correct any conclusions resting on 308 if it's wrong)

Existing reference files (read-only for this diagnostic):
- `bucket_brigade/envs/bucket_brigade_env.py:110–155` — step() and action shape
- `bucket_brigade/envs/scenarios_generated.py:70–85` — `default_scenario(num_agents)`
- `bucket_brigade/envs/puffer_env.py:77` — MultiDiscrete([10, 2]) confirmation
- `experiments/p3_specialization/analyze_plateau.py:48–52` — current hard-coded baselines
- Issue #145 body — original 4012.34/episode measurement
- PR #183 / issue #174 — training cells using the 308 acceptance bar

### Test Plan

- [ ] Manual verification: run `uv run python experiments/p3_specialization/diagnostics/random_baseline.py` locally (single-machine, ~1 minute — pure env stepping, no training; this IS safe to run locally per the CLAUDE.md compute guidelines)
- [ ] Verify reported mean falls within plausible bounds (200–400 per step team reward — sanity)
- [ ] Verify CI width tightens with more episodes (200 ep × 5 seeds should give ±2–5 reward per step)
- [ ] Cross-check: the original #145 measurement used 50 episodes — re-run with 50 episodes (single seed) and confirm we land near 308 ± noise; this validates we're using the same env API as #145

### Scope Notes

- **In scope**: re-derive `default` random baseline at `num_agents=4`; reconcile with 290 iter-0 and 308 cited; document verdict.
- **Out of scope** (per issue body, kept): re-running PPO training; computing a "fixed-good-strategy" baseline; other scenarios beyond `default`.
- **Estimated effort**: 20 min (script is ~30 LOC, runs in ~1 min, write-up is small)

---

📝 **Curator note**: Located full provenance chain for "308" (originated in #145 issue body, hard-coded in `analyze_plateau.py:48–52` and `analyze_174.py:47`, no committed measurement script). Verified env API matches issue assumptions with one correction: episode length is `min_nights=12` + termination delay, not a fixed 13.
